SIGSEGV in ioxsql tests after upgrade to Arrow v7

#14 broke the `ioxsql` tests. This package's tests depend on a running instance of IOx, so it's hard to catch failures like this. I'll see if a newer version of arrow fixes the issue.

```
--- FAIL: TestNormalLifeCycle (0.03s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x10279603c]
...
github.com/influxdata/influxdb-iox-client-go/ioxsql.(*rows).Next(0x14000110180, {0x1400012c240, 0x2, 0x2})
        /Users/jacobmarble/projects/influxdb-iox-client-go/ioxsql/driver_rows.go:73 +0x70
...
github.com/influxdata/influxdb-iox-client-go/ioxsql_test.TestNormalLifeCycle(0x140002fe820)
        /Users/jacobmarble/projects/influxdb-iox-client-go/ioxsql/driver_test.go:106 +0x22c
...
FAIL    github.com/influxdata/influxdb-iox-client-go/ioxsql     0.262s
FAIL
```

_Originally posted by @jacobmarble in https://github.com/influxdata/influxdb-iox-client-go/issues/14#issuecomment-1064592858_

--- a/iox_client_query.go
+++ b/iox_client_query.go
@@ -64,9 +64,10 @@ type QueryRequest struct {
 
 func newRequest(client *Client, database, query string) *QueryRequest {
 	return &QueryRequest{
-		client:   client,
-		database: database,
-		query:    query,
+		client:    client,
+		database:  database,
+		query:     query,
+		allocator: memory.DefaultAllocator,
 	}
 }
 

